### PR TITLE
Feature/issue 764 tweak build deploy script

### PR DESCRIPTION
**Pull request by rasmus-storjohann-PG**

### CODE REVIEW

**Comment by rasmus-storjohann-PG** [comment]
@ian-lin-PG @tomy-pg please review

File: bin/prepare_deploy.sh [hunk 1]
```diff
@@ -272,36 +257,30 @@ setStagingValuesInAppJson() {
...
 
     if [ "$BUILD" == "staging" ]
     then
-        echo "GOOGLE_ANALYTICS_TRACKING_ID='UA-30770107-5'" >> "$CLIENT_DIRECTORY/.env"
-        echo "API_URL=$STAGING_URL"                         >> "$CLIENT_DIRECTORY/.env"
-
         setStagingValuesInAppJson
 
+        cp ../pathways-deploy/staging/env .env
+        cp ../pathways-deploy/staging/google-services.json .
```
Comment (tomy-pg) [inline bin/prepare_deploy.sh:275]: This should be 
```
cp ../pathways-deploy/staging/env $CLIENT_DIRECTORY/.env
cp ../pathways-deploy/staging/google-services.json $CLIENT_DIRECTORY
```
Without these changes, we will are copying those environment variables and the json file to our local directory, whereas we want it in the working directory.

File: bin/prepare_deploy.sh [hunk 2]
```diff
@@ -272,36 +257,30 @@ setStagingValuesInAppJson() {
...
+        setVersionInEnv
+
     elif [ "$BUILD" == "production" ]
     then
-        echo "GOOGLE_ANALYTICS_TRACKING_ID='UA-30770107-3'" >> "$CLIENT_DIRECTORY/.env"
-        echo "API_URL=$PRODUCTION_URL"                      >> "$CLIENT_DIRECTORY/.env"
+        cp ../pathways-deploy/production/env .env
+        cp ../pathways-deploy/production/google-services.json .
```
Comment (tomy-pg) [inline bin/prepare_deploy.sh:281]: See comment above.

File: bin/prepare_deploy.sh [hunk 3]
```diff
@@ -13,10 +13,6 @@ while (( "$#" )); do
     then
         SERVER_VERSION=$2
         shift 2
-    elif [ "$1" == "--androidVersionCode" ]
-    then
-        ANDROID_VERSION_CODE=$2
```
Comment (tomy-pg) [inline bin/prepare_deploy.sh:18]: This change affects `validateClientVersion()` as well. The following lines
```
validateClientVersion() {
...
FILE_CODE=$(grep versionCode "$CLIENT_DIRECTORY/app.json")
    if [ "$FILE_CODE" != "      \"versionCode\": $ANDROID_VERSION_CODE," ]
    then
        echo "Error: client app.json contains $FILE_CODE when $ANDROID_VERSION_CODE was expected"
        fail
    fi
```
should be removed or the release process exits.

**Comment by rasmus-storjohann-PG** [comment]
Nice, thanks for catching those issues

File: bin/prepare_deploy.sh (hunk 3 above)
Comment (rasmus-storjohann-PG) [reply bin/prepare_deploy.sh:18]: I checked the unit tests, they will still catch this kind of error, so the validation in the prepare deploy script was actually redundant.

**Pull request merged**